### PR TITLE
feat(Discover): Refactoring Truncation in Legend

### PR DESCRIPTION
**Pull request by ayesha-omarali**
Made so the props are the same as how we input truncation in echart's tooltip.

### CODE REVIEW

File: src/sentry/static/sentry/app/components/charts/baseChart.jsx [hunk 1]
```diff
@@ -54,9 +54,6 @@ class BaseChart extends React.Component {
     // Chart legend
     legend: SentryTypes.EChartsLegend,
```
Comment (billyvg) [inline src/sentry/static/sentry/app/components/charts/baseChart.jsx]: Let's add the `truncate` option in here

**Review by billyvg — APPROVED** [review]

File: src/sentry/static/sentry/app/components/charts/baseChart.jsx (hunk 1 above)
Comment (ayesha-omarali) [reply src/sentry/static/sentry/app/components/charts/baseChart.jsx]: How come? I noticed TooltipOptions doesn't include it in there, but I do enforce it in SentryTypes if that makes up for it? Also truncate isn't used anywhere in baseChart.jsx

Comment (billyvg) [reply src/sentry/static/sentry/app/components/charts/baseChart.jsx]: Look at `SentryTypes.EchartsTooltip`

File: src/sentry/static/sentry/app/sentryTypes.jsx [hunk 2]
```diff
@@ -685,6 +685,7 @@ export const EChartsGrid = PropTypes.shape({
 export const EChartsLegend = PropTypes.shape({
   // Show legend on chart
   show: PropTypes.bool,
+  truncate: PropTypes.oneOfType([PropTypes.bool, PropTypes.number]),
```
Comment (lynnagara) [inline src/sentry/static/sentry/app/sentryTypes.jsx]: `truncate` doesn't currently support boolean, though it'd be good if it did apply a default length (same as tooltip)

Comment (ayesha-omarali) [reply src/sentry/static/sentry/app/sentryTypes.jsx]: added it 👍

**Pull request merged**